### PR TITLE
Adds function `best_case_worst_case_avg_reduction_targets()` to create columns `avg_reduction_targets_best_case` and `avg_reduction_targets_worst_case`

### LINKED ISSUE #290: Add `reduction_target_avg_best_case` & `reduction_target_avg_worst_case` on company-level
Dear @kalashsinghal , 

following the same pattern as in this ticket (2DegreesInvesting/tiltIndicatorAfter/issues/286), please add the columns `reduction_target_avg_best_case `& `reduction_target_avg_worst_case `on company-level, reflecting the best & worst case values for the average sector reduction targets per company. This is required for the webtool data. 

Thanks!

### PR DESCRIPTION
**Pull request by kalashsinghal**
closes #290 

Dear @Tilmon @AnneSchoenauer @maurolepore ,

This PR adds the `avg_reduction_targets_best_case` and `avg_reduction_targets_worst_case` at company level for transition risk profile indicator using the function `best_case_worst_case_avg_reduction_targets()`. 

The methodology for this function is same as the methodology used for `best_case_worst_case_avg_profile_ranking()` function in this [PR](https://github.com/2DegreesInvesting/tiltIndicatorAfter/pull/293).

Reprex for review:

``` r
library(readr, warn.conflicts = FALSE)
library(dplyr, warn.conflicts = FALSE)
library(tiltToyData, warn.conflicts = FALSE)
library(tiltTransitionRisk, warn.conflicts = FALSE)
devtools::load_all(".")
#> ℹ Loading tiltIndicatorAfter
options(width = 5000)

set.seed(123)
restore <- options(list(
  readr.show_col_types = FALSE,
  tiltIndicatorAfter.output_co2_footprint = TRUE
))

toy_emissions_profile_products_ecoinvent <- read_csv(toy_emissions_profile_products_ecoinvent()) 
toy_emissions_profile_any_companies <- read_csv(toy_emissions_profile_any_companies())
toy_sector_profile_any_scenarios <- read_csv(toy_sector_profile_any_scenarios())
toy_sector_profile_companies <- read_csv(toy_sector_profile_companies())
toy_europages_companies <- read_csv(toy_europages_companies())
toy_ecoinvent_activities <- read_csv(toy_ecoinvent_activities())
toy_ecoinvent_europages <- read_csv(toy_ecoinvent_europages())
toy_ecoinvent_inputs <- read_csv(toy_ecoinvent_inputs())
toy_isic_name <- read_csv(toy_isic_name())
toy_all_activities_scenario_sectors <- read_csv(toy_all_activities_scenario_sectors())

toy_emissions_profile <- profile_emissions(
  companies = toy_emissions_profile_any_companies,
  co2 = toy_emissions_profile_products_ecoinvent,
  europages_companies = toy_europages_companies,
  ecoinvent_activities = toy_ecoinvent_activities,
  ecoinvent_europages = toy_ecoinvent_europages,
  isic = toy_isic_name
)
toy_sector_profile <- profile_sector(
  companies = toy_sector_profile_companies,
  scenarios = toy_sector_profile_any_scenarios,
  europages_companies = toy_europages_companies,
  ecoinvent_activities = toy_ecoinvent_activities,
  ecoinvent_europages = toy_ecoinvent_europages,
  isic = toy_isic_name
)

output <- transition_risk_profile(
  emissions_profile = toy_emissions_profile,
  sector_profile = toy_sector_profile,
  co2 = toy_emissions_profile_products_ecoinvent,
  all_activities_scenario_sectors = toy_all_activities_scenario_sectors,
  scenarios = toy_sector_profile_any_scenarios,
  pivot_wider = TRUE
)

product <- output |> 
  unnest_product() |>
  filter(companies_id == "nonphilosophical_llama",
         scenario %in% c("1.5C RPS", "NZ 2050"),
         benchmark %in% c("all"))

product |> 
  print(n = Inf)
#> # A tibble: 8 × 39
#>   companies_id           company_name           country postcode address                                main_activity ep_product                  matched_activity_name                                    matched_reference_product                     matching_certainty co2_footprint unit  isic_4digit tilt_sector tilt_subsector benchmark co2e_lower co2e_upper profile_ranking emission_profile emissions_profile_equal_weight emissions_profile_best_case emissions_profile_worst_case scenario  year reduction_targets sector_profile benchmark_tr_score transition_risk_score transition_risk_low_threshold transition_risk_high_threshold transition_risk_category transition_risk_profile_equal_weight transition_risk_profile_best_case transition_risk_profile_worst_case amount_of_distinct_products min_headcount max_headcount activity_uuid_product_uuid          
#>   <chr>                  <chr>                  <chr>   <chr>    <chr>                                  <chr>         <chr>                       <chr>                                                    <chr>                                         <chr>                      <dbl> <chr> <chr>       <chr>       <chr>          <chr>          <dbl>      <dbl>           <dbl> <chr>                                     <dbl>                       <dbl>                        <dbl> <chr>    <dbl>             <dbl> <chr>          <chr>                              <dbl>                         <dbl>                          <dbl> <chr>                                                   <dbl>                             <dbl>                              <dbl>                       <int>         <dbl>         <dbl> <chr>                               
#> 1 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface finishing, galvanic market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 1.5C RPS  2030              0.09 low            1.5C RPS_2030_all                  0.212                         0.366                          0.492 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 2 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface finishing, galvanic market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 1.5C RPS  2050              0.95 high           1.5C RPS_2050_all                  0.642                         0.756                          0.872 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 3 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface finishing, galvanic market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 NZ 2050   2030              0.22 medium         NZ 2050_2030_all                   0.277                         0.391                          0.532 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 4 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface finishing, galvanic market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 NZ 2050   2050              0.96 high           NZ 2050_2050_all                   0.647                         0.751                          0.864 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 5 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface engineering         market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 1.5C RPS  2030              0.09 low            1.5C RPS_2030_all                  0.212                         0.366                          0.492 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 6 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface engineering         market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 1.5C RPS  2050              0.95 high           1.5C RPS_2050_all                  0.642                         0.756                          0.872 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 7 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface engineering         market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 NZ 2050   2030              0.22 medium         NZ 2050_2030_all                   0.277                         0.391                          0.532 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b
#> 8 nonphilosophical_llama nonphilosophical_llama germany 35606    industriegebiet oberbiel | 35606 solms distributor   surface engineering         market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode low                        0.693 kg    '2591'      metals      other metals   all           -0.106       1.91           0.333 low                                         0.5                         0.5                          0.5 NZ 2050   2050              0.96 high           NZ 2050_2050_all                   0.647                         0.751                          0.864 low                                                       0.5                               0.5                                0.5                           2             1            10 833caa78-30df-4374-900f-7f88ab44075b

company <- output |> 
  unnest_company() |>
  filter(companies_id == "nonphilosophical_llama",
         scenario %in% c("1.5C RPS", "NZ 2050"),
         benchmark %in% c("all"))

company |> 
  print(n = Inf)
#> # A tibble: 4 × 34
#>   companies_id           company_name           country co2_avg benchmark profile_ranking_avg avg_profile_ranking_best_case avg_profile_ranking_worst_case emission_category_low emission_category_medium emission_category_high emission_category_NA scenario  year reduction_targets_avg avg_reduction_targets_best_case avg_reduction_targets_worst_case sector_category_low sector_category_medium sector_category_high sector_category_NA benchmark_tr_score_avg avg_transition_risk_equal_weight avg_transition_risk_best_case avg_transition_risk_worst_case postcode address                                main_activity min_headcount max_headcount transition_risk_category_high transition_risk_category_medium transition_risk_category_low transition_risk_category_NA
#>   <chr>                  <chr>                  <chr>     <dbl> <chr>                   <dbl>                         <dbl>                          <dbl>                 <dbl>                    <dbl>                  <dbl>                <dbl> <chr>    <dbl>                 <dbl>                           <dbl>                            <dbl>               <dbl>                  <dbl>                <dbl>              <dbl> <chr>                                             <dbl>                         <dbl>                          <dbl> <chr>    <chr>                                  <chr>                 <dbl>         <dbl>                         <dbl>                           <dbl>                        <dbl>                       <dbl>
#> 1 nonphilosophical_llama nonphilosophical_llama germany   0.693 all                     0.333                         0.333                          0.333                     1                        0                      0                    0 1.5C RPS  2030                  0.09                            0.09                             0.09                   1                      0                    0                  0 1.5C RPS_2030_all                                 0.212                         0.212                          0.212 35606    industriegebiet oberbiel | 35606 solms distributor               1            10                             0                               0                            1                           0
#> 2 nonphilosophical_llama nonphilosophical_llama germany   0.693 all                     0.333                         0.333                          0.333                     1                        0                      0                    0 1.5C RPS  2050                  0.95                            0.95                             0.95                   0                      0                    1                  0 1.5C RPS_2050_all                                 0.642                         0.642                          0.642 35606    industriegebiet oberbiel | 35606 solms distributor               1            10                             0                               0                            1                           0
#> 3 nonphilosophical_llama nonphilosophical_llama germany   0.693 all                     0.333                         0.333                          0.333                     1                        0                      0                    0 NZ 2050   2030                  0.22                            0.22                             0.22                   0                      1                    0                  0 NZ 2050_2030_all                                  0.277                         0.277                          0.277 35606    industriegebiet oberbiel | 35606 solms distributor               1            10                             0                               0                            1                           0
#> 4 nonphilosophical_llama nonphilosophical_llama germany   0.693 all                     0.333                         0.333                          0.333                     1                        0                      0                    0 NZ 2050   2050                  0.96                            0.96                             0.96                   0                      0                    1                  0 NZ 2050_2050_all                                  0.647                         0.647                          0.647 35606    industriegebiet oberbiel | 35606 solms distributor               1            10                             0                               0                            1                           0
```

<sup>Created on 2024-07-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR description to reflect what it ended up doing.
- [x] Polish the PR title as you'd like others to read it from the git log.
- [x] Assign a reviewer.
- [ ] Document user-facing changes in the changelog.

### CODE REVIEW

File: R/best_case_worst_case_avg_reduction_targets.R [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    add_avg_reduction_targets_best_case() |>
+    add_avg_reduction_targets_worst_case()
+}
+
+add_avg_reduction_targets <- function(data) {
+  data |>
+    add_avg_col("avg_reduction_targets",
+                "reduction_targets",
+                "scenario_year",
+                col_sector_profile())
```
Comment (maurolepore) [inline R/best_case_worst_case_avg_reduction_targets.R:64]: The style is a lil different than in the tidyverse style guide.

File: R/best_case_worst_case_avg_reduction_targets.R [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+add_avg_reduction_targets_best_case <- function(data) {
+  data |>
+    add_avg_case_col_if_risk_category_match(
+      "avg_reduction_targets_best_case",
+      col_sector_profile(),
+      "min_risk_category_per_company_benchmark",
+      "avg_reduction_targets"
+    )
```
Comment (maurolepore) [inline R/best_case_worst_case_avg_reduction_targets.R:74]: This style IS consistent with the tidyverse style guide.

File: tests/testthat/test-best_case_worst_case_avg_reduction_targets.R [hunk 3]
```diff
@@ -0,0 +1,74 @@
+test_that("`NA` in `sector_profile` and `reduction_targets` gives correct values for `avg_reduction_targets_best_case` and `avg_reduction_targets_worst_case`", {
+  example_data <- example_best_case_worst_case_reduction_targets_product_level(
+    sector_profile = c("low", "low", NA_character_),
+    reduction_targets = c(1.0, 2.0, NA)
+  )
+
+  out <- create_avg_reduction_targets_best_case_worst_case_at_product_level(example_data)
+
+  expected_best_case <- c(3 / 2, NA)
+  expected_worst_case <- c(3 / 2, NA)
```
Comment (maurolepore) [inline tests/testthat/test-best_case_worst_case_avg_reduction_targets.R:10]: Do we expect the same for both cases?

**Review by maurolepore — DISMISSED** [review]
LGTM. 

I feel I already reviewed this PR or one very similar. If so maybe there is some potential to DRY some parts.

Thanks.

File: tests/testthat/test-best_case_worst_case_avg_reduction_targets.R (hunk 3 above)
Comment (kalashsinghal) [reply tests/testthat/test-best_case_worst_case_avg_reduction_targets.R:10]: Yes, we expect the same for both cases

**Pull request merged**